library/src/hipsolver_module.f90

```diff
@@ -2957,16 +2963,217 @@ function hipsolverZheevd(handle, jobz, uplo, n, A, lda, D, work, lwork, info) &
             integer(c_int), value :: n
             type(c_ptr), value :: A
             integer(c_int), value :: lda
-            type(c_ptr), value :: D
+            type(c_ptr), value :: W
             type(c_ptr), value :: work
             integer(c_int), value :: lwork
             type(c_ptr), value :: info
         end function hipsolverZheevd
     end interface
 
+    ! ******************** SYEVDX/HEEVDX ********************
+    interface
+        function hipsolverSsyevdx_bufferSize(handle, jobz, range, uplo, n, A, lda, vl, vu, il, iu, nev, W, lwork) &
+                result(c_int) &
```

Comment:
We've been declaring our return type incorrectly in the Fortran bindings. The `result` declaration takes a variable name, not a type. The right way of doing this would be `result(status)` and then below doing `integer(int_c), value :: status` to the argument list. Or, well, almost. The return value isn't necessarily `c_int` anyway. The type should really be `integer(kind(HIPSOLVER_STATUS_SUCCESS))`. There is also an alternative syntax for the return type that doesn't use `result` at all. You'd just put `integer(kind(HIPSOLVER_STATUS_SUCCESS))` before `function`.

Fixing this one instance would be rather pointless, so I don't think you need to change anything here. However, I don't want to let it go without comment. We really should create a concrete plan to address this issue.

I think it's pretty clear that we're not Fortran experts. I wonder if we might now be able to deprecate this interface so you don't need to keep updating it. The hipFORT project provides these bindings too, and it's maintained by developers with a better understanding of Fortran.

Reply:
Yeah, I'm inclined to let hipFORT take over the Fortran bindings for us. We might want to discuss it in our upcoming meeting. I wish we had a sense for if anyone is using these Fortran bindings right now.

Reply:
I've opened https://github.com/ROCmSoftwarePlatform/hipSOLVER/pull/105 to fix the existing API functions. Could you update the added syevdx functions to match?

Reply:
Sure, once the other PR is merged in.